lib/common/output_log.c

```diff
@@ -0,0 +1,267 @@
+/*
+ * Copyright 2019 the Pacemaker project contributors
+ *
+ * The version control history for this file may have further details.
+ *
+ * This source code is licensed under the GNU Lesser General Public License
+ * version 2.1 or later (LGPLv2.1+) WITHOUT ANY WARRANTY.
+ */
+#ifndef _GNU_SOURCE
+#  define _GNU_SOURCE
+#endif
+
+#include <ctype.h>
+#include <libxml/HTMLtree.h>
+#include <stdarg.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <crm/crm.h>
+#include <crm/common/output.h>
+#include <crm/common/xml.h>
+
+GOptionEntry pcmk__log_output_entries[] = {
+    { NULL }
+};
+
+typedef struct private_data_s {
+    /* gathered in log_begin_list */
+    GQueue/*<char*>*/ *prefixes;
+} private_data_t;
+
+static void
+log_subprocess_output(pcmk__output_t *out, int exit_status,
+                      const char *proc_stdout, const char *proc_stderr) {
+    /* This function intentionally left blank */
+}
+
+static void
+log_free_priv(pcmk__output_t *out) {
+    private_data_t *priv = out->priv;
+
+    if (priv == NULL) {
+        return;
+    }
+    
+    g_queue_free(priv->prefixes);
+    free(priv);
+}
+
+static bool
+log_init(pcmk__output_t *out) {
+
+    /* If log_init was previously called on this output struct, just return. */
+    if (out->priv != NULL) {
+        return true;
+    }
+
+    out->priv = calloc(1, sizeof(private_data_t));
+    if (out->priv == NULL) {
+         return false;
+    }
+    ((private_data_t *)out->priv)->prefixes = g_queue_new();
+    return true;
+}
+
+static void
+log_finish(pcmk__output_t *out, crm_exit_t exit_status, bool print, void **copy_dest) {
+    /* This function intentionally left blank */
+}
+
+static void
+log_reset(pcmk__output_t *out) {
+    CRM_ASSERT(out && out->priv);
+
+    log_free_priv(out);
+    log_init(out);
+}
+
+G_GNUC_PRINTF(2, 3)
+static void
+pcmk__output_crm_log(pcmk__output_t *out, const char *format, ...) {
+    int len = 0;
+    va_list ap;
+    char *buffer = NULL;
+
+    va_start(ap, format);
+    len = vasprintf(&buffer, format, ap);
+    CRM_ASSERT(len >= 0);
+    va_end(ap);
+
+    qb_log_from_external_source(NULL, NULL, "%s", LOG_INFO, 0, 0, buffer);
```

Comment:
Seeing it reduced, I realize we don't even need pcmk__output_crm_log(), we can just use crm_info() directly

Reply:
Done.